### ISSUE TITLE
What is needed to make this production ready?

### ISSUE THREAD

**Issue by jayarjo** [mavlink/MAVSDK-JavaScript#40]

**Comment by JonasVautherin** [comment]
It's just a proof of concept, so quite some work. Are you willing to contribute, or are you wondering if that will be ready soon? Happy to get contributions, but otherwise, nobody is working on this at the moment...

**Comment by jayarjo** [comment]
Yes. I guess it might turn out easier, than relearning Python, unless that list is too long... I'd probably convert whole thing to TypeScript first.

Is the following list still actual?

1. Write a nice API for some selected features (e.g. action.arm, action.takeoff, and telemetry.position) using RxJS.
2. (Optional) Make a small UI using those features (e.g. a "takeoff" button, and show the position values somewhere).
3. Write templates based on 1) to auto-generate the full API from our [proto](https://github.com/mavlink/MAVSDK-JavaScript/blob/main/proto/protos) definitions.
4. Deploy the package.

**Comment by JonasVautherin** [comment]
Yes :+1:

**Comment by antonymarion** [comment]
@jayarjo @JonasVautherin any news on this?

It looks like this repo has been abandoned for a while, hasn't it ?

Or maybe there is some MavSDK in TS somewhere and I did not search in the right direction ^^, let me know in this case

**Comment by JonasVautherin** [comment]
I think it's pretty much dead at this point 😅. Unless someone wants to revive it (and probably restart it from scratch?)

**Comment by antonymarion** [comment]
> I think it's pretty much dead at this point 😅. Unless someone wants to revive it (and probably restart it from scratch?)

I will, depending on my company needs.

Since both ts and mavlink are widely used.
Thus there is gap here in MavSDK for those in the Web using SAAS solutions who want their solution to be fully coded in TS.

Do you see any tricky difficulty for doing this lib?

Here are some advices from Claude:
https://mammouth.ai/shared/95038138-6ef5-4bd1-86b2-77bf92d9c3d7
